integrated_channels/sap_success_factors/client.py

```diff
@@ -31,28 +33,119 @@ def get_oauth_access_token(url_base, client_id, client_secret, company_id, user_
                 'scope': {
                     'userId': user_id,
                     'companyId': company_id,
-                    'userType': 'user',
+                    'userType': 'admin',
                     'resourceType': 'learning_public_api',
                 }
             },
-            auth=(client_id, client_secret)
+            auth=(client_id, client_secret),
+            headers={'content-type': 'application/json'}
         )
 
+        response.raise_for_status()
         data = response.json()
-
         try:
-            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expiresIn'])
-        except KeyError:
+            return data['access_token'], datetime.datetime.utcfromtimestamp(data['expires_in'] + int(time.time()))
+        except Exception:
             raise requests.RequestException(response=response)
+
+    def __init__(self, enterprise_configuration):
+        """
+        Instantiate a new client.
+
+        Args:
+            enterprise_configuration (SAPSuccessFactorsEnterpriseCustomerConfiguration): An enterprise customers's
+            configuration model for connecting with SAP SuccessFactors
+
+        Raises:
+            ValueError: If a URL or access token are not provided.
+        """
+
+        if not enterprise_configuration:
+            raise ValueError('An SAPSuccessFactorsEnterpriseCustomerConfiguration must be supplied!')
+
+        self.enterprise_configuration = enterprise_configuration
+        self._create_session()
+        super(SAPSuccessFactorsAPIClient, self).__init__()
+
+    def _create_session(self):
+        """
+        Instantiate a new session object for use in connecting with SAP SuccessFactors
+        """
+        session = requests.Session()
+        session.timeout = self.SESSION_TIMEOUT
+
+        oauth_access_token, expires_at = SAPSuccessFactorsAPIClient.get_oauth_access_token(
+            self.enterprise_configuration.sapsf_base_url,
+            self.enterprise_configuration.key,
+            self.enterprise_configuration.secret,
+            self.enterprise_configuration.sapsf_company_id,
+            self.enterprise_configuration.sapsf_user_id
+        )
+
+        session.headers['Authorization'] = 'Bearer {}'.format(oauth_access_token)
+        self.session = session
+        self.expires_at = expires_at
+
+    def send_completion_status(self, payload):
+        """
+        Send a completion status payload to the SuccessFactors OCN Completion Status endpoint
+
+        Args:
+            payload: JSON object (serialized from LearnerDataTransmissionAudit)
+                containing completion status fields per SuccessFactors documentation.
+
+        Returns:
+            The body of the response from SAP SuccessFactors, if successful
+        Raises:
+            HTTPError: if we received a failure response code from SAP SuccessFactors
+        """
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
```

Comment:
It appears we are using this in both of the "public" functions for the class -- does it make sense to DRY it by relocating to `__init__() as a `self` field so we don't have to remember to include in new functions going forward?